modules/atomic-widgets/template-renderer/template-renderer.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace Elementor\Modules\AtomicWidgets\TemplateRenderer;
+
+use Elementor\Utils;
+use ElementorDeps\Twig\Environment;
+use ElementorDeps\Twig\TemplateWrapper;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+class Template_Renderer {
+	private static ?self $instance = null;
+
+	private Single_File_Loader $loader;
+
+	private Environment $env;
+
+	private function __construct() {
+		$this->loader = new Single_File_Loader();
+
+		$this->env = new Environment(
+			$this->loader,
+			[
+				'debug' => Utils::is_elementor_debug(),
+				'autoescape' => 'name',
+			]
+		);
+	}
+
+	public static function instance(): self {
+		if ( ! isset( self::$instance ) ) {
```

Comment:
```suggestion
		if ( null === self::$instance ) {
```

maybe if you wanna be explicit

Reply:
Good point, Done.